code/modules/surgery/implant_removal.dm

```diff
@@ -1,81 +1,199 @@
 /datum/surgery/implant_removal
-	name = "Implant removal"
+	name = "Implant extraction/destruction"
 	possible_locs = list(BODY_ZONE_CHEST)
 	steps = list(
 		/datum/surgery_step/incise,
 		/datum/surgery_step/clamp_bleeders,
 		/datum/surgery_step/retract_skin,
-		/datum/surgery_step/extract_implant,
+		/datum/surgery_step/search_for_implants,
+		/datum/surgery_step/destroy_extract_implant,
 		/datum/surgery_step/close,
 	)
+	var/list/discovered_implants = list()
 
-//extract implant
-/datum/surgery_step/extract_implant
-	name = "extract implant (hemostat)"
+/datum/surgery_step/search_for_implants
+	name = "search for implants (hand/t-ray)"
+	accept_hand = TRUE
 	implements = list(
-		TOOL_HEMOSTAT = 100,
-		TOOL_CROWBAR = 65,
-		/obj/item/kitchen/fork = 35)
-	time = 64
-	success_sound = 'sound/surgery/hemostat1.ogg'
-	var/obj/item/implant/implant
+		/obj/item/t_scanner = 100,
+	)
+	time = 12.8 SECONDS
+	repeatable = TRUE
+	var/detection_prob = list(
+		/obj/item/t_scanner = 20,
+		/obj/item/healthanalyzer/advanced = 15,
+		/obj/item/healthanalyzer = 10,
+		"" = 5, // empty string is no tool, so hand.
+	)
```

Comment:
Having a roundstart engineering tool be better than head of staff/researched medical gear? That's just silly. Construction tools are generally the ghetto option and I think it should stay that way. Health analyzers are professional medical tools, the t-ray is for looking at pipes. You should make the ranking for this:

1. Advanced Health Analyzer = 20
2. Health Analyzer = 15
3. T-Ray Scanner = 10
4. Empty Hand = 5

Reply:
not a code review

Reply:
Seems as valid as a review complaining about how long something takes, but ok.